examples/notebooks/creating_a_problem.ipynb

```diff
@@ -0,0 +1,194 @@
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "5XU-dMtU04p2"
+   },
+   "source": [
+    "### Creating a Model\n",
+    "\n",
+    "A fitting problem consists of three parts: An experimental dataset that is used for fitting, a model that the dataset should be fit to, and the parameters that should be fit. To obtain a problem we therefore first need to create a model. This has been explained in detail in the creating_a_model example. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "parameter_set = pybop.ParameterSet.pybamm(\"Marquis2019\")\n",
+    "model = pybop.lithium_ion.SPM(parameter_set=parameter_set)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Next, the parameters. All parameters of the problem are contained in pybop.Parameters(). Each of the variable to fit is defined as a pybop.Parameter() separated by commas inside pybop.Parameters(). In this example, we have two parameters, the positive electrode thickness and the positive particle radius:\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "parameters = pybop.Parameters(\n",
+    "    pybop.Parameter(\n",
+    "        \"Positive electrode thickness [m]\",\n",
+    "        prior=pybop.Gaussian(7.56e-05, 0.05e-05),\n",
+    "        bounds=[65e-06, 10e-05],\n",
+    "    ),\n",
+    "    pybop.Parameter(\n",
+    "        \"Positive particle radius [m]\",\n",
+    "        prior=pybop.Gaussian(5.22e-06, 0.05e-06),\n",
+    "        bounds=[2e-06, 9e-06],\n",
+    "    ),\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The third component is the dataset. This could be an experimental dataset or a simulation. In this example, we just use simulation data by running model.predict() for a 1C discharge, similar to the creating_a_problem example. To pass it to the problem, we need to extract the voltage, time and current data from the simulation results and define it as the dataset:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "t_eval = np.arange(0, 3700, 1)\n",
+    "solution = model.predict([], t_eval)  # No inputs i.e []\n",
+    "dataset = pybop.Dataset(\n",
+    "    {\n",
+    "        \"Time [s]\": np.arange(0, len(solution[\"Voltage [V]\"].data), 1),\n",
+    "        \"Current function [A]\": solution[\"Current [A]\"].data,\n",
+    "        \"Voltage [V]\": solution[\"Voltage [V]\"].data,\n",
+    "    }\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Setting up the problem\n",
+    "With all parts setup, we can now make a problem. Therefore, we use pybop.FittingProblem() to define a simple fitting problem and the model, parameters and dataset defined before as inputs."
```

Comment:
Can you also expand this to include the `pybop.DesignProblem`? I think we can reduce the cost / optimisation side of this and focus on the two problem classes more.